### ISSUE TITLE
Add debug log entry for message received and handler executed

### ISSUE THREAD

**Issue by SimonCropp** [Particular/NServiceBus#5179]
Currently: 

> When NServiceBus sends a message, it writes the result of the ToString() method of the message class to the log

https://docs.particular.net/nservicebus/logging/message-contents

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Pipeline/Outgoing/RoutingToDispatchConnector.cs#L44

I would have assumed that it would be equally (or perhaps more so) helpful to have log entries for message received and handler executed?

or am i missing something and this is already implemented somewhere?

**Comment by andreasohlund** [comment]
Pretty sure we don't have that, you mean to DEBUG log

```
Processing message type: {The Message Type}

Headers: {the headers}
Body: {the body}

Handlers to invoke: Handler1 => Handler2 ... => HandlerN

```

**Comment by SimonCropp** [comment]
@andreasohlund yep, along those lines

**Comment by andreasohlund** [comment]
Makes sense, @Particular/nservicebus-maintainers I've flagged this as a maintenance release candidate, let me know if you  have concerns

**Comment by andreasohlund** [comment]
Fair point but since we already log the bodies of outgoing messages this is a problem we already have?

If so should we consider doing something about it?

**Comment by SimonCropp** [comment]
while i dont agree that this violates GDPR, the startup logging would also contain similar information

**Comment by andreasohlund** [comment]
Pretty sure we can since we use the same call and order when we're loading up the handlers and kicks of the invocation part of the pipeline

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs#L27

**Comment by SimonCropp** [comment]
@timbussmann thanks
